libs/ml/include/ml/core/graph.hpp

```diff
@@ -938,6 +940,15 @@ std::vector<TensorType> Graph<TensorType>::GetGradientsReferences() const
   return ret;
 }
 
+template <typename TensorType>
```

Comment:
function comment please. Please describe why this function exists and when we might invoke it.
it's not obvious why you get a vector of unordered sets here unless you already have the word2vec application in mind.